### ISSUE TITLE
[Bug]: HTTP Error: grpc stream error: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (7833328 vs. 4194304)

### ISSUE THREAD

**Issue by SOF3** [jaegertracing/jaeger#4350]
### What happened?

As a user with large traces, I want to configure the grpc plugin max receive message length so that jaeger can display large traces (e.g. with more than 10000 spans).

### Steps to reproduce

1. Send a large trace response from a grpc storage plugin
2. The server returns the error

### Expected behavior

The server should allow increasing the maximum message size in parity to collector

### Relevant log output

```shell
{
  "level": "error",
  "ts": 1680074638.9349482,
  "caller": "app/http_handler.go:495",
  "msg": "HTTP handler, Internal Server Error",
  "error": "grpc stream error: rpc error: code = ResourceExhausted desc = grpc: received message larger than max (7833328 vs. 4194304)",
  "stacktrace": "github.com/jaegertracing/jaeger/cmd/query/app.(*APIHandler).handleError\n\tgithub.com/jaegertracing/jaeger/cmd/query/app/http_handler.go:495\ngithub.com/jaegertracing/jaeger/cmd/query/app.(*APIHandler).getTrace\n\tgithub.com/jaegertracing/jaeger/cmd/query/app/http_handler.go:437\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2110\ngithub.com/opentracing-contrib/go-stdlib/nethttp.MiddlewareFunc.func5\n\tgithub.com/opentracing-contrib/go-stdlib@v1.0.0/nethttp/server.go:154\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2110\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2110\ngithub.com/gorilla/mux.(*Router).ServeHTTP\n\tgithub.com/gorilla/mux@v1.8.0/mux.go:210\ngithub.com/jaegertracing/jaeger/cmd/query/app.additionalHeadersHandler.func1\n\tgithub.com/jaegertracing/jaeger/cmd/query/app/additional_headers_handler.go:28\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2110\ngithub.com/gorilla/handlers.CompressHandlerLevel.func1\n\tgithub.com/gorilla/handlers@v1.5.1/compress.go:100\nnet/http.HandlerFunc.ServeHTTP\n\tnet/http/server.go:2110\ngithub.com/gorilla/handlers.recoveryHandler.ServeHTTP\n\tgithub.com/gorilla/handlers@v1.5.1/recovery.go:78\nnet/http.serverHandler.ServeHTTP\n\tnet/http/server.go:2951\nnet/http.(*conn).serve\n\tnet/http/server.go:1992"
}
```

### Screenshot

<img width="1391" alt="image" src="https://user-images.githubusercontent.com/19623715/228479739-2d291346-0876-4c2d-b05f-cbf9f39a6455.png">

### Jaeger backend version

v1.42

### SDK

custom storage plugin

### Pipeline

In a nutshell, a custom storage plugin that mutates the trace.

See https://github.com/kubewharf/kelemetry/blob/14752b56543603fe3c84bc817cb437ede51fe1d6/docs/DEPLOY.md for full pipeline, but it is mostly irrelevant.

### Stogage backend

custom storage backend

### Operating system

Linux

### Deployment model

Kubernetes

### Deployment configs

```shell
N/A
```

**Comment by yurishkuro** [comment]
Which storage implementation are you using?  The message limit is already 4Mb _for the grpc-client running in jaeger-query_. So while technically it is possible to increase its max message size, the client is already using grpc streaming to allow larger volumes of data to be passed. In other words, your remote storage that implements the Jaeger remote storage API could be configured to return results in smaller chunks (<4Mb).

But having said that, I don't object with having a configuration in Jaeger too.

**Comment by SOF3** [comment]
I am using a private storage implementation that doesn't collect data through jaeger-collector. I am just using jaeger as a display frontend.

**Comment by jkowall** [comment]
Should be fixed in Jaeger v2 when we have moved to otel, feel free to re-open if not.

**Comment by chenlujjj** [comment]
Hi @yurishkuro , I've met a same issue when implementing v2 grpc remote storage API.

The workaround I'm using now is to set `grpc.MaxCallRecvMsgSize(20*1024*1024)` in client code like:
```
stream, err := tr.client.GetTraces(ctx, &storage.GetTracesRequest{
			Query: query,
		}, grpc.MaxCallRecvMsgSize(20*1024*1024))
```
(The change is made in `internal/storage/v2/grpc/tracereader.go` file.)

>In other words, your remote storage that implements the Jaeger remote storage API could be configured to return results in smaller chunks (<4Mb)

I'm not quite sure how to do that. For example, when implementing `GetTraces(req *storage.GetTracesRequest, stream storage.TraceReader_GetTracesServer) error` method, I need to call `stream.Send(&v1.TracesData{ResourceSpans: spans})`, how to split the `TracesData` into smaller chunks making sure each chunk is <4MB?

**Comment by yurishkuro** [comment]
TracesData is a wrapper with a collection inside, why can't you split it?

**Comment by chenlujjj** [comment]
@yurishkuro I can split it, the problem is how to make sure the the size is below 4MB ? I don't see a `Size()` like method of `TracesData`. The built-in `unsafe.Sizeof()` method result may not be the same of proto buffer size.
